ports/mimxrt/hal/board.h

```diff
@@ -0,0 +1,16 @@
+/* Dummy file - necessary for compilation with NXP MCU SDK
+ *  Based on NXP MCU SDK /project_template/board.h
+ *
+ * Copyright 2019 NXP
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
```

Comment:
This file is pretty much written from scratch, so I think you should use your own license/copyright.  And it's not a dummy file because it includes things that are actually needed!